pinot-core/src/main/java/org/apache/pinot/core/segment/store/SingleFileIndexDirectory.java

```diff
@@ -258,15 +255,15 @@ private void mapBufferEntries()
       IndexEntry entry = offsetEntry.getValue();
       runningSize += entry.size;
 
-      if (runningSize >= MAX_ALLOCATION_SIZE) {
+      if (runningSize >= MAX_ALLOCATION_SIZE && !offsetAccum.isEmpty()) {
```

Comment:
So with this condition, the first two indices will be loaded in one larray. Right?

Reply:
Without this check, if the size of the first index is over 2G, it will throw exception because `offsetAccum` is empty.

Reply:
I understand that, but with this check, we will combine the first two indices together into one larray mmap. May be that is not what we want.

Reply:
No, that won't happen. It will work the same as before but preventing the exception. When the second index come, we will load the first index, and the second index will get into the next file

Reply:
Agreed.